crates/bevy_text/src/pipeline.rs

```diff
@@ -0,0 +1,120 @@
+use std::collections::HashMap;
+
+use ab_glyph::{PxScale, ScaleFont};
+use bevy_asset::{Assets, Handle, HandleId};
+use bevy_math::Size;
+use bevy_render::prelude::Texture;
+use bevy_sprite::TextureAtlas;
+
+use glyph_brush_layout::{FontId, SectionText};
+
+use crate::{
+    error::TextError, glyph_brush::GlyphBrush, Font, FontAtlasSet, TextAlignment, TextVertex,
+};
+
+pub struct TextPipeline {
+    pub brush: GlyphBrush,
+    pub map_font_id: HashMap<HandleId, FontId>,
+}
+
+impl Default for TextPipeline {
+    fn default() -> Self {
+        let brush = GlyphBrush::default();
+        let map_font_id = HashMap::default();
+        TextPipeline { brush, map_font_id }
+    }
+}
+
+impl TextPipeline {
+    pub fn measure(
+        &mut self,
+        font_handle: Handle<Font>,
+        font_storage: &Assets<Font>,
+        text: &str,
+        scale: f32,
+        text_alignment: TextAlignment,
+        bounds: Size,
+    ) -> Result<Size, TextError> {
+        let font = font_storage
+            .get(font_handle.id)
+            .ok_or(TextError::NoSuchFont)?;
+
+        let font_id = self.get_or_insert_font_id(font_handle, font);
+
+        let section = SectionText {
+            font_id,
+            scale: PxScale::from(scale),
+            text,
+        };
+
+        let scaled_font = ab_glyph::Font::as_scaled(&font.font, scale);
+
+        let section_glyphs = self
+            .brush
+            .compute_glyphs(&[section], bounds, text_alignment)?;
+
+        if section_glyphs.is_empty() {
+            return Ok(Size::new(0., 0.));
+        }
+        let first_glyph = section_glyphs.first().unwrap();
+        let mut min_x: f32 =
+            first_glyph.glyph.position.x - scaled_font.h_side_bearing(first_glyph.glyph.id);
+        let mut min_y: f32 = first_glyph.glyph.position.y - scaled_font.ascent();
+        let mut max_x: f32 =
+            first_glyph.glyph.position.x + scaled_font.h_advance(first_glyph.glyph.id);
+        let mut max_y: f32 = first_glyph.glyph.position.y - scaled_font.descent();
```

Comment:
we can remove redundant code here by replacing this section with f32::MAX/MIN values:

```rust
let mut min_x = std::f32::MAX;
let mut min_y = std::f32::MAX;
let mut max_x = std::f32::MIN;
let mut max_y = std::f32::MIN;
```